src/org/daisy/dotify/formatter/impl/row/SegmentProcessor.java

```diff
@@ -1115,51 +1114,38 @@ private String getLeftIndentWithLabel(String listLabel, int indent) {
         private Optional<RowImpl> continueRow(
             RowInfo row,
             BrailleTranslatorResult btr,
-            int nextRowIndent,
             int rightIndentIfNotLastRow,
             String mode,
             LineProperties lineProps
         ) {
-            RowImpl ret = null;
+            if (!hasCurrentRow()) {
+                // continueRow() should only be called after having checked that there is a current
+                // row, or after having created a new one.
+                throw new RuntimeException("Error in code.");
```

Comment:
OK I can see where you're coming from, but it's just meant as an assertion. The message is not meant as a user-level error message. I guess I could change it into an `assert`, but the reason I usually not do that is because assertions may be disabled.

Reply:
I added "Personally" because it is partly a matter of taste. "Error in code" is just so meaningless. I've encountered this message several times - and there were cases where I think I didn't do anything wrong. Of course you can still determine what went wrong by checking the stack trace.

Reply:
Yes, checking the stack trace and the source is the idea.

If you read up on using assertions, they also recommend not providing a message unless "the program has some additional information that might help diagnose the failure" (see https://docs.oracle.com/javase/7/docs/technotes/guides/language/assert.html).

If one of these error messages ever gets exposed to users because of a bug, I think something like "Coding error" makes it clear that they have to contact the developers. Any additional information can only be confusing for the user because it is not meant for them.

> I've encountered this message several times - and there were cases where I think I didn't do anything wrong.

What do you mean?